lib/deprecation.nix

```diff
@@ -46,6 +46,21 @@ rec {
         '';
     };
 
+  processRenameOptions =
+    options: oldPath: newPath:
+    (map (
+      option:
+      let
+        result =
```

Comment:
`let result; in result` is redundant (unless you're re-implementing `lib.fix`), I think we can drop the outer let block here?